### PR TITLE
Support loading DMI data from DT systems

### PR DESCRIPTION
**Pull request by hughsie**
To do this with the existing codebase just fake the required SMBIOS structures.

### CODE REVIEW

File: libfwupdplugin/fu-smbios.c [hunk 1]
```diff
@@ -64,12 +64,71 @@ typedef struct __attribute__((packed)) {
...
+	fu_smbios_convert_dt_string (self, FU_SMBIOS_STRUCTURE_TYPE_SYSTEM, 0x04,
+				     path, "vendor");
+
+	/* DMI:Family */
+	fu_smbios_convert_dt_string (self, FU_SMBIOS_STRUCTURE_TYPE_SYSTEM, 0x1a,
+				     path, "model-name");
+
+	/* DMI:ProductName */
+	fu_smbios_convert_dt_string (self, FU_SMBIOS_STRUCTURE_TYPE_SYSTEM, 0x05,
+				     path, "model");
```
Comment (superm1) [inline libfwupdplugin/fu-smbios.c:115]: on a DT system do these also populate `/sys/class/dmi/id`?  If so, maybe it's better to just fallback to that path in fu-smbios.

Comment (hughsie) [reply libfwupdplugin/fu-smbios.c:115]: On my qemu system there's no `/sys/class/dmi` whatsoever. @meklort is this populated on actual hardware please?

Comment (superm1) [reply libfwupdplugin/fu-smbios.c:115]: if nothing brings devicetree stuff like this to /sys/class/dmi on real hardware then yes agree this PR looks fine

Comment (meklort) [reply libfwupdplugin/fu-smbios.c:115]: > On my qemu system there's no `/sys/class/dmi` whatsoever. @meklort is this populated on actual hardware please?

Correct, this does not exist on the Talos II or Blackbird systems:

    meklort@talos2 ~/g/b/build ((e8e9e8df…))> ls /sys/class/dmi
    ls: cannot access '/sys/class/dmi': No such file or directory

**Pull request merged**